### PR TITLE
changed results tables in online covariance to sycl for gpu

### PR DESCRIPTION
**Pull request by mchernov-intel**
@vepifanov

### CODE REVIEW

**Comment by mchernov-intel** [comment]
make test

File: algorithms/kernel/covariance/covariance_partialresult.h [hunk 1]
```diff
@@ -47,9 +48,21 @@ DAAL_EXPORT services::Status PartialResult::allocate(const daal::algorithms::Inp
...
+
+    if (deviceInfo.isCpu)
+    {
+        set(nObservations, HomogenNumericTable<size_t>::create(1, 1, NumericTable::doAllocate, &status));
+        set(crossProduct, HomogenNumericTable<algorithmFPType>::create(nColumns, nColumns, NumericTable::doAllocate, &status));
+        set(sum, HomogenNumericTable<algorithmFPType>::create(nColumns, 1, NumericTable::doAllocate, &status));
+    }
+    else
+    {
+        set(nObservations, SyclHomogenNumericTable<size_t>::create(1, 1, NumericTable::doAllocate, &status));
```
Comment (PetrovKP) [inline algorithms/kernel/covariance/covariance_partialresult.h]: Why create a SyclHomogenNumericTable 1 x 1 ?

File: algorithms/kernel/covariance/covariance_partialresult.h [hunk 2]
```diff
@@ -47,9 +48,21 @@ DAAL_EXPORT services::Status PartialResult::allocate(const daal::algorithms::Inp
...
 
-    set(nObservations, HomogenNumericTable<size_t>::create(1, 1, NumericTable::doAllocate, &status));
-    set(crossProduct, HomogenNumericTable<algorithmFPType>::create(nColumns, nColumns, NumericTable::doAllocate, &status));
-    set(sum, HomogenNumericTable<algorithmFPType>::create(nColumns, 1, NumericTable::doAllocate, &status));
+    auto & context    = services::Environment::getInstance()->getDefaultExecutionContext();
+    auto & deviceInfo = context.getInfoDevice();
+
+    if (deviceInfo.isCpu)
+    {
+        set(nObservations, HomogenNumericTable<size_t>::create(1, 1, NumericTable::doAllocate, &status));
```
Comment (PetrovKP) [inline algorithms/kernel/covariance/covariance_partialresult.h]: NumericTable is support `size_t` type?

Comment (rlnx) [reply algorithms/kernel/covariance/covariance_partialresult.h]: You can create homogen table with `size_t`, but you cannot read it with generic NT interface 😅

**Comment by rlnx** [comment]
@PetrovKP or @vepifanov, can you run precommit again to make sure everything is ok? I think, it can be merged if CI is green.

**Comment by PetrovKP** [comment]
> @PetrovKP or @vepifanov, can you run precommit again to make sure everything is ok? I think, it can be merged if CI is green.

CI has failed on tests of online covariance for GPU.

**Comment by rlnx** [comment]
@mchernov-intel, why it is still not merged?

**Comment by mchernov-intel** [comment]
clone of https://github.com/intel/daal/pull/262

**Pull request closed without merging**